es-core/src/Scripting.cpp

```diff
@@ -0,0 +1,52 @@
+#include "Scripting.h"
+#include "Log.h"
+#include "platform.h"
+#include "utils/FileSystemUtil.h"
+
+namespace Scripting
+{
+	int fireEvent(const std::string& eventName, const std::string& arg1, const std::string& arg2)
+	{
+		LOG(LogDebug) << "fireEvent: " << eventName << " " << arg1 << " " << arg2;
+
+		std::list<std::string> scriptDirList;
+		std::string test;
+
+		// check in exepath
+		test = Utils::FileSystem::getExePath() + "/scripts/" + eventName;
+		if(Utils::FileSystem::exists(test))
+			scriptDirList.push_back(test);
+
+		// check in homepath
+		test = Utils::FileSystem::getHomePath() + "/.emulationstation/scripts/" + eventName;
+		if(Utils::FileSystem::exists(test))
+			scriptDirList.push_back(test);
+
+		// loop over found script paths per event and over scripts found in eventName folder.
+		for(std::list<std::string>::const_iterator dirIt = scriptDirList.cbegin(); dirIt != scriptDirList.cend(); ++dirIt) {
+			std::list<std::string> scripts = Utils::FileSystem::getDirContent(*dirIt);
+			for (std::list<std::string>::const_iterator it = scripts.cbegin(); it != scripts.cend(); ++it) {
+#ifndef WIN32 // osx / linux
+				if (!Utils::FileSystem::isExecutable(*it)) {
+					LOG(LogWarning) << *it << " is not executable. Did you 'chmod u+x'?. Skipping this script.";
+					continue;
+				}
+#endif
+				std::string script = *it;
+				if (arg1.length() > 0) {
+					script += " \"" + arg1 + "\"";
+					if (arg2.length() > 0) {
+						script += " \"" + arg2 + "\"";
+					}
+				}
+				LOG(LogDebug) << "  executing: " << script;
+				int rc = runSystemCommand(script);
+				if (rc != 0) {
```

Comment:
I'm not sure if I like this change.  I envisioned the scripting kind of like an event dispatcher, fire and forget.  This changes that.  Mainly concerned that one script failing could prevent others from executing.  Of course, the user could/should handle any errors in the script to prevent that from happening.  Just wanted to open the discussion and see what others think.